### PR TITLE
 PTI-238: Add orders

### PR DESCRIPTION
**Pull request by danieltruong**

### CODE REVIEW

File: angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.html [hunk 1]
```diff
@@ -27,8 +27,10 @@ <h4 class="blue-header">Shared Data [Master]</h4>
           <div class="row">
             <div class="col-7"></div>
             <div class="col-5">
-              <span class="grey-subtext">If content is published on any site, hitting the Save/Update
-                Record button will update the public site immediately</span>
+              <span class="grey-subtext"
+                >If content is published on any site, hitting the Save/Update Record button will update the public site
```
Comment (marklise) [inline angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.html]: Put the > on the line above

Comment (NickPhura) [reply angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.html]: I dont know all of the ins and outs of spans, but I believe the reason for the > on the lower line is to ensure that intentional spaces that might come before the text aren't confused with the indentation spaces from the code formatting.

File: angular/tslint.json [hunk 2]
```diff
@@ -89,7 +89,7 @@
     "no-submodule-imports": false,
     "no-switch-case-fall-through": true,
     "no-trailing-whitespace": true,
-    "no-unused-expression": true,
+    "no-unused-expression": false,
```
Comment (NickPhura) [inline angular/tslint.json]: Why unset this setting?

If there are variables, etc, that are completely unused, its best to remove them so they don't  clutter the code unnecessarily.

File: angular/projects/global/src/lib/utils/utils.ts [hunk 3]
```diff
@@ -22,4 +30,90 @@ export class Utils {
...
+        .replace(/\%2F/g, '_')
+        .replace(/ /g, '_');
+    } else {
+      return filename
+        .replace(/\(/g, '%28')
+        .replace(/\)/g, '%29')
+        .replace(/\\/g, '_')
+        .replace(/\//g, '_');
+    }
+  }
```
Comment (NickPhura) [inline angular/projects/global/src/lib/utils/utils.ts]: Is there any reason why we wouldn't just use something like: 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

to handle encoding?

File: angular/projects/global/src/lib/utils/utils.ts [hunk 4]
```diff
@@ -22,4 +30,90 @@ export class Utils {
...
+        month = '0' + month;
+      }
+      if (day.length < 2) {
+        day = '0' + day;
+      }
+      return [year, month, day].join('');
+    } else {
+      return null;
+    }
+  }
```
Comment (NickPhura) [inline angular/projects/global/src/lib/utils/utils.ts]: This seems like the kind of stuff MomentJS is good at.  Or is there a reason to do it manually ourselves?

File: angular/projects/global/src/lib/utils/utils.ts [hunk 5]
```diff
@@ -1,3 +1,11 @@
+const encode = encodeURIComponent;
+window['encodeURIComponent'] = (component: string) => {
+  return encode(component).replace(/[!'()*]/g, c => {
+    // Also encode !, ', (, ), and *
+    return '%' + c.charCodeAt(0).toString(16);
+  });
+};
+
```
Comment (NickPhura) [inline angular/projects/global/src/lib/utils/utils.ts]: Similar question further below.

Any reason not to use something like: 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

to handle encoding?

File: angular/tslint.json (hunk 2 above)
Comment (danieltruong) [reply angular/tslint.json]: This was causing short circuiting to fail lint

File: angular/projects/global/src/lib/utils/utils.ts (hunk 4 above)
Comment (marklise) [reply angular/projects/global/src/lib/utils/utils.ts]: This should be tossed as it's not being used anywhere.

File: angular/projects/global/src/lib/utils/utils.ts (hunk 3 above)
Comment (marklise) [reply angular/projects/global/src/lib/utils/utils.ts]: Toss this as it's not being used.

File: angular/tslint.json (hunk 2 above)
Comment (NickPhura) [reply angular/tslint.json]: Apparently they added a setting to account for this: https://palantir.github.io/tslint/rules/no-unused-expression/

So we just need to change it to:

```"no-unused-expression": [true, "allow-fast-null-checks"]```

**Pull request merged**